problem2/problem2.cpp

```diff
@@ -0,0 +1,34 @@
+class FooBar {
+private:
+    int n;
+    mutex m;
+    int state = 0;
+    condition_variable cv;
+
+    void execute(function<void()> print, int whichTurn) {
+        unique_lock<mutex> lck(m);
```

Comment:
You're not releasing this lock until you've iterated through the entire for loop, I think this will lead to a deadlock.